clients/src/main/java/org/apache/kafka/clients/admin/DescribeReplicaLogDirResult.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.kafka.common.requests.DescribeLogDirsResponse;
+
+
+/**
+ * The result of {@link AdminClient#describeReplicaLogDir(Collection)}.
+ *
+ * The API of this class is evolving, see {@link AdminClient} for details.
+ */
+@InterfaceStability.Evolving
+public class DescribeReplicaLogDirResult {
+    private final Map<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> futures;
+
+    DescribeReplicaLogDirResult(Map<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> futures) {
+        this.futures = futures;
+    }
+
+    /**
+     * Return a map from replica to future which can be used to check the log directory information of individual replicas
+     */
+    public Map<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> values() {
+        return futures;
+    }
+
+    /**
+     * Return a future which succeeds if log directory information of all replicas are available
+     */
+    public KafkaFuture<Map<TopicPartitionReplica, ReplicaLogDirInfo>> all() {
+        return KafkaFuture.allOf(futures.values().toArray(new KafkaFuture[0])).
+            thenApply(new KafkaFuture.Function<Void, Map<TopicPartitionReplica, ReplicaLogDirInfo>>() {
+                @Override
+                public Map<TopicPartitionReplica, ReplicaLogDirInfo> apply(Void v) {
+                    Map<TopicPartitionReplica, ReplicaLogDirInfo> replicaLogDirInfos = new HashMap<>();
+                    for (Map.Entry<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> entry : futures.entrySet()) {
+                        try {
+                            replicaLogDirInfos.put(entry.getKey(), entry.getValue().get());
+                        } catch (InterruptedException | ExecutionException e) {
+                            // This should be unreachable, because allOf ensured that all the futures completed successfully.
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    return replicaLogDirInfos;
+                }
+            });
+    }
+
+    static public class ReplicaLogDirInfo {
+        public final String currentReplicaLogDir;
+        public final String temporaryReplicaLogDir;
+        public final long temporaryReplicaOffsetLag;
+
+        public ReplicaLogDirInfo() {
+            this(null, null, DescribeLogDirsResponse.INVALID_OFFSET_LAG);
+        }
+
+        public ReplicaLogDirInfo(String currentReplicaLogDir, String temporaryReplicaLogDir, long temporaryReplicaOffsetLag) {
+            this.currentReplicaLogDir = currentReplicaLogDir;
+            this.temporaryReplicaLogDir = temporaryReplicaLogDir;
+            this.temporaryReplicaOffsetLag = temporaryReplicaOffsetLag;
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder builder = new StringBuilder();
+            if (temporaryReplicaLogDir != null) {
+                builder.append("ReplicaLogDirInfo(currentReplicaLogDir=")
```

Comment:
I think we usually do not print the field name of a map value, we usually not not print the name field of the value. Otherwise shouldn't we print the field name of the key as well? So for maps, it would simply be [key, value]. In this case, each entry could be printed as something like [topic-0-1, (currentLogDir=XXX, temporaryLogDir=YYY, temporaryReplicaOffsetLag=ZZZ)], which is probably clear enough.

BTW, I forgot to comment that I think we should add documentation about what are the value combination of those fields mean. For example is it possible I get a currentLogDir=null, tempLogDir=XXX, offsetsLage=-1? If so what does that mean.